compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -348,84 +348,55 @@ class ExpressionCodegen(
         }
 
-        val receiver = expression.dispatchReceiver
-        receiver?.apply {
+        expression.dispatchReceiver?.let { receiver ->
             callGenerator.genValueAndPut(
                 callee.dispatchReceiverParameter!!,
-                this,
+                receiver,
                 if (isSuperCall) receiver.asmType else callable.dispatchReceiverType
                     ?: throw AssertionError("No dispatch receiver type: ${expression.render()}"),
-                this@ExpressionCodegen,
+                this,
                 data
             )
         }
 
-        expression.extensionReceiver?.apply {
+        expression.extensionReceiver?.let { receiver ->
             callGenerator.genValueAndPut(
                 callee.extensionReceiverParameter!!,
-                this,
+                receiver,
                 callable.extensionReceiverType!!,
-                this@ExpressionCodegen,
+                this,
                 data
             )
         }
 
         callGenerator.beforeValueParametersStart()
-        val typeParameters = if (callee is IrConstructor)
-            callee.parentAsClass.typeParameters + callee.typeParameters
-        else
-            callee.typeParameters
-        val typeArguments = (0 until typeParameters.size).map { expression.getTypeArgument(it)!! }
-        val typeSubstitutionMap = typeParameters.map { it.symbol }.zip(typeArguments).toMap()
         expression.symbol.owner.valueParameters.forEachIndexed { i, irParameter ->
             val arg = expression.getValueArgument(i)
             val parameterType = callable.valueParameterTypes[i]
-            when {
-                arg != null -> {
-                    callGenerator.genValueAndPut(irParameter, arg, parameterType, this@ExpressionCodegen, data)
-                }
-                else -> {
-                    assert(irParameter.varargElementType != null)
-                    val type = typeMapper.mapType(
-                        irParameter.type.substitute(typeSubstitutionMap)
-                    )
-                    callGenerator.putValueIfNeeded(
-                        parameterType,
-                        StackValue.operation(type) {
-                            it.aconst(0)
-                            it.newarray(correctElementType(type))
-                        },
-                        ValueKind.GENERAL_VARARG, i, this@ExpressionCodegen
-                    )
-                }
-            }
+            require(arg != null) { "Null argument in ExpressionCodegen for parameter ${irParameter.render()}" }
+            callGenerator.genValueAndPut(irParameter, arg, parameterType, this, data)
         }
 
         expression.markLineNumber(true)
 
         // Do not generate redundant markers in continuation class.
-        if (callee.isSuspend && !irFunction.isInvokeSuspendInContinuation()) {
+        if (callee.isSuspend && !irFunction.isInvokeSuspendInContinuation())
```

Comment:
We prefer to keep the braces in such cases. (Makes it harder to accidentally botch the scope of the conditional.)

Reply:
Done!